ivy/exceptions.py

```diff
@@ -56,15 +60,43 @@ def new_fn(*args, **kwargs):
         try:
             return fn(*args, **kwargs)
         except (IndexError, ValueError, AttributeError) as e:
-            if ivy.get_exception_trace_mode():
-                raise ivy.exceptions.IvyError(fn.__name__, str(e))
+            if (
+                ivy.get_exception_trace_mode() == "frontend"
```

Comment:
we shouldn't repeatedly call `ivy.get_exception_trace_mode()`. We should call this once and then store the result in a variable called `trace_mode`.